libs/core/src/lib/file-uploader/file-uploader.component.ts

```diff
@@ -179,15 +180,24 @@ export class FileUploaderComponent implements ControlValueAccessor {
         }
         const maxSize = this.parseFileSize(this.maxFileSize);
         const minSize = this.parseFileSize(this.minFileSize);
+
         const checkSize = (fileSize: number): boolean => {
-            if (this.maxFileSize && fileSize > maxSize) { return false; }
-            if (this.minFileSize && fileSize < minSize) { return false; }
+            if (this.maxFileSize && fileSize > maxSize) {
+                return false;
+            }
+            if (this.minFileSize && fileSize < minSize) {
+                return false;
+            }
             return true;
-        }
+        };
 
-        this.validFiles = event.filter(file => checkSize(file.size));
-        this.invalidFiles = event.filter(file => !checkSize(file.size));
+        let allowedExtensions = null;
+        if (this.accept) {
+            allowedExtensions = this.accept.toLocaleLowerCase().replace(/[\s.]/g, '').split(',')
+        }
 
+        this.validFiles = event.filter(file => checkSize(file.size) && this._checkExtension(file, allowedExtensions));
+        this.invalidFiles = event.filter(file => !checkSize(file.size) || !this._checkExtension(file, allowedExtensions));
```

Comment:
should we not change style, if invalid file is getting uploaded.
like we do in other form component, we set state = 'error' or 'warning'.

Reply:
For sure state shouldn't be changed under the hood

Reply:
I tried to add some example with states, but `readonly` state of input has it's own styling and it overwrites the semantic states(warning, success etc.). We should contact designer how it should look like in this case